spec/solver_spec.rb

```diff
@@ -0,0 +1,24 @@
+require_relative '../solver'
+describe Solver do
+  context 'When passed an input' do
+    solver = Solver.new
+    it 'expect factorial of 5 tobe 120' do
+      expect(solver.factorial(5)).to eq 120
+    end
+    it 'expect positive number ' do
+      expect(solver.factorial(5)).to be >= 0
+    end
```

Comment:
- [ ] Please also make sure to handle the special cases of the #factorial method - when input is 0, the result should equal 1